tensorboard/components/tf_ng_tensorboard/core/core.effects.test.ts

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {TestBed} from '@angular/core/testing';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import {Actions} from '@ngrx/effects';
+import {Action} from '@ngrx/store';
+import {provideMockActions} from '@ngrx/effects/testing';
+import {ReplaySubject} from 'rxjs';
+
+import {CoreEffects} from './core.effects';
+import * as coreActions from './core.actions';
+import {CoreService} from './core.service';
+import {PluginsListing} from '../types/api';
+
+import * as _typeHackEffects from '@ngrx/effects';
```

Comment:
Huh, super weird. Apparently, I didn't need these hacks for this file because I happen to have write import. From another file (core.actions.ts), it results in something like below: 
```
tensorboard/components/tf_ng_tensorboard/core/core.actions.ts(33,14): error TS2742: The inf
erred type of 'pluginsListingFailed' cannot be named without a reference to '../../../../ex
ternal/npm/node_modules/@ngrx/store/store'. This is likely not portable. A type annotation
is necessary.     
```

This make no sense to me....